### PR TITLE
Dajung

### PR DESCRIPTION
**Pull request by ss01363**
player setup
treasure setup
field view button setup

### CODE REVIEW

File: TreasureHunt/Assets/2.Scripts/Hunter/Buttons/MenuButtonController.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		myButton = GetComponent<Button> ();
+		myButton.onClick.AddListener (GotIt);
+	}
+
+	void Start(){
+		target.SetActive (false);
+	}
+
+	void GotIt(){
+		if (flag) {
```
Comment (cvanderw) [inline TreasureHunt/Assets/2.Scripts/Hunter/Buttons/MenuButtonController.cs:22]: This code seems slightly redundant. I think you can do without the flags.

Notice that GameObject has an [activeSelf](https://docs.unity3d.com/ScriptReference/GameObject-activeSelf.html) field.

What about the following:

```
void GotIt() {
    target.SetActive(!target.activeSelf);
}
```

File: TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+using SimpleJSON;
+
+public class TreasureSetupController : MonoBehaviour {
+
+	public static TreasureSetupController instance = null;
+
+	void Awake()
+	{
+		if (instance == null)
+
```
Comment (cvanderw) [inline TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs:14]: There's a few unnecessary blank lines here. Also, you sometimes put braces around conditional (if) statements and sometimes don't. Try to be consistent.

File: TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+		else if (instance != this)
+
+			Destroy(gameObject);
+
+		DontDestroyOnLoad (gameObject);
+
+	}
+
+	void Start(){
```
Comment (cvanderw) [inline TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs:25]: Your brace style differs from Hyun's. You guys need to get together and figure out what your coding convention is :).

File: TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+		GameObject gameTreasurePanel = new GameObject();
+		gameTreasurePanel.tag = "GameTreasurePanel";
+
+		for (int i = 0; i < games.Count; i++) {
+			//make new game panel
+			GameObject newGame = (GameObject) Instantiate (game, new Vector3(0,0,0), Quaternion.identity);
+			GameAttributes ga = newGame.GetComponent<GameAttributes> ();
+			// attach game attributes
+			ga.SetAsChildOf(gameTreasurePanel);
+			ga.SetAttributes(games[i]["game_id"], games[i]["game_name"], games[i]["treasure_count"].AsInt, games[i]["maker_id"], games[i]["status"].AsInt);
```
Comment (cvanderw) [inline TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs:68]: There's a whole bunch of `games[i]` going on here. What about pulling this out into a single variable to simplify some of this?

**Review by cvanderw — COMMENTED** [review]
I left a couple of stylistic comments. I understand you guys have a demo coming up so I won't block your merge.

**Comment by hyunSo** [comment]
I resolved conflict first for demonstration today. Merge first then let's edit later after today's stand-up.

File: TreasureHunt/Assets/2.Scripts/Hunter/Buttons/MenuButtonController.cs (hunk 1 above)
Comment (ss01363) [reply TreasureHunt/Assets/2.Scripts/Hunter/Buttons/MenuButtonController.cs:22]: Thank you! Fixed it! ><

File: TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs (hunk 2 above)
Comment (ss01363) [reply TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs:14]: Fixed it before! Thanks X)

File: TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs (hunk 4 above)
Comment (ss01363) [reply TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs:68]: Fixed it! Thank you~

**Pull request merged**